RepoMan/RepoMan/Repository/RepoManagerFactory.cs

```diff
@@ -0,0 +1,42 @@
+using System;
+using System.Threading.Tasks;
+using RepoMan.IO;
+using RepoMan.Records;
+using Serilog;
+
+namespace RepoMan.Repository
+{
+    public class RepoManagerFactory :
+        IRepoManagerFactory
+    {
+        private readonly IPullRequestReaderFactory _prReaderFactory;
+        private readonly IPullRequestCacheManager _cacheManager;
+        private readonly TimeSpan _prApiDosBuffer;
+        private readonly ILogger _logger;
+
+        public RepoManagerFactory(IPullRequestReaderFactory prReaderFactory, IPullRequestCacheManager cacheManager, TimeSpan prApiDosBuffer, ILogger logger)
+        {
+            _prReaderFactory = prReaderFactory ?? throw new ArgumentNullException(nameof(prReaderFactory));
+            _cacheManager = cacheManager ?? throw new ArgumentNullException(nameof(cacheManager));
+            _prApiDosBuffer = prApiDosBuffer < TimeSpan.Zero
+                ? throw new ArgumentOutOfRangeException("TimeSpan must be greater than or equal to zero")
+                : prApiDosBuffer;
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        }
+
+        public async Task<IRepoManager> GetManagerAsync(WatchedRepository repo, bool refreshFromUpstream)
+        {
+            var prReader = _prReaderFactory.GetReader(repo);
+            var manager = await RepositoryManager.InitializeAsync(
+                repo.Owner,
+                repo.Name,
+                repo.Url,
+                prReader,
+                _cacheManager,
+                _prApiDosBuffer,
+                refreshFromUpstream,
+                _logger);
+            return manager;
+        }
+    }
```

Comment:
All these bazillion factories may seem a bit Java-circa-1999-ish, but I swear it's the cleanest and most maintainable way of handling what to do at the `WatchedRepository` level, which is also the right level of abstraction to handle things at.

Reply:
I had the exact same thought -- it's the cleanest way to have a single composition root and the only thing you have to think about to use the tool is setting up the watched repo.